File: tests/fixtures/sample.go
package main

type Config struct {
	Name    string
	Verbose bool
	Count   int
}

type Reader interface {
	Read(p []byte) (n int, err error)
	Close() error
}

func NewConfig(name string) *Config {
	return &Config{Name: name}
}

func (c *Config) IsVerbose() bool {
	return c.Verbose
}

func (c *Config) SetCount(n int) {
	c.Count = n
}

func add(a, b int) int {
	return a + b
}

File: tests/fixtures/sample.rs
/// A simple struct.
pub struct Config {
    pub name: String,
    pub verbose: bool,
}

/// An enum for status.
enum Status {
    Active,
    Inactive,
}

/// A trait for display.
trait Renderable {
    fn render(&self) -> String;
}

/// Top-level function.
pub fn greet(name: &str) -> String {
    format!("Hello, {}!", name)
}

/// Private helper.
fn add(a: i32, b: i32) -> i32 {
    a + b
}

impl Config {
    pub fn new(name: String) -> Self {
        Config {
            name,
            verbose: false,
        }
    }

    fn is_verbose(&self) -> bool {
        self.verbose
    }
}

type Pair = (i32, i32);
